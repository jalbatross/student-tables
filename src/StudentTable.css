.table-title {
    text-align:center;


.ReactTable .rt-thead {
  overflow-y: scroll;
}}

.current-grade {
    color: blue;
    text-align:center;
}
.finished-grade {
    color:green;
    background-color:yellow;
    width:100%;
    height:100%;
    text-align:center;

}

.cell-styling {
    border-width: thin;
    border-color:black;
    border-style: solid;
    padding:0 !important;
    border-right:0 !important;
    border-bottom:1px !important;

}