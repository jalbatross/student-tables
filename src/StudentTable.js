import React, { Component } from "react";
import ReactDOM from 'react-dom';
import ReactTable from 'react-table';
import 'react-table/react-table.css';

class StudentTable extends React.Component {
    constructor(props) {
        super(props);
    }

    parseData(data) {
        console.log(data);
        if (data.toString().indexOf('/') !== -1) {
            var dateFormat = require('dateformat');
            var date = new Date(data);
            return React.createElement(
                'span',
                { className: 'date' },
                dateFormat(date, "shortDate").toString()
            );
        } else {
            return React.createElement(
                'span',
                { className: 'score' },
                data.toString()
            );
        }
    }

    render() {

        const columns = [{
            Header: 'Student Name',
            accessor: 'name' // String-based value accessors!
        }, {
            Header: 'Grade 2',
            accessor: 'g2_score',
            Cell: props => React.createElement(
                'span',
                { className: 'score' },
                ' ',
                props.value,
                ' '
            ) // Custom cell components!
        }, {
            Header: 'Grade 3',
            accessor: 'g3_score',
            Cell: props => this.parseData(props.value)
        }];

        return React.createElement(ReactTable, { data: this.props.data, columns: columns });
    }

}

export default StudentTable;

