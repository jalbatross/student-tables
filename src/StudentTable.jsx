import React, { Component }  from "react";
import ReactDOM from 'react-dom';
import ReactTable from 'react-table';
import 'react-table/react-table.css';

class StudentTable extends React.Component {
    constructor(props) {
        super(props);
    }

    parseData(data) {
        console.log(data)
        if (data.toString().indexOf('/') !== -1) {
            var dateFormat = require('dateformat');
            var date = new Date(data);
            return <span className='date'>{dateFormat(date, "shortDate").toString()}</span>;
        }
        else {
            return <span className='score'>{data.toString()}</span>;
            
        }
    }

    render() {

        const columns = [{
            Header: 'Student Name',
            accessor: 'name' // String-based value accessors!
        }, {
            Header: 'Grade 2',
            accessor: 'g2_score',
            Cell: props => < span className = 'score' > { props.value } < /span> // Custom cell components!
        }, {
            Header: 'Grade 3',
            accessor: 'g3_score',
            Cell: props => this.parseData(props.value)
        }]
        
        return(
            <ReactTable data={this.props.data} columns={columns}/> 
        )
    }


}

export default StudentTable;