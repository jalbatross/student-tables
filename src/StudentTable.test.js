import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import StudentTable from './StudentTable';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() }); 

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<StudentTable data={[]} columns={[]} title={""} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const component = renderer.create(
        <StudentTable data={[]} columns = {[]} title={''} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correct title', () => {
    const component = shallow(<StudentTable title ={'hello'}/>);
    expect(component.find('.table-title')).toHaveLength(1);
    expect(component.find('.table-title').text()).toBe('hello');
});

it('applies correct cell formatting', () => {
  
});
