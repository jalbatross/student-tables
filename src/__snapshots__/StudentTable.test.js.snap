// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span
    className="table-title ReactTable"
  >
    
  </span>
  <div
    className="ReactTable"
    style={
      Object {
        "height": "400px",
      }
    }
  >
    <div
      className="rt-table"
      role="grid"
      style={undefined}
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "0px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
          style={undefined}
        />
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "0px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -odd"
            role="row"
            style={Object {}}
          />
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -padRow -even"
            role="row"
            style={Object {}}
          />
        </div>
      </div>
    </div>
    <div
      className="rt-noData"
    >
      No rows found
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
